src/storage/v2/storage.cpp

```diff
@@ -1928,8 +1954,14 @@ utils::BasicResult<Storage::RegisterReplicaError> Storage::RegisterReplica(
   });
 }
 
-bool Storage::UnregisterReplica(const std::string_view name) {
+bool Storage::UnregisterReplica(const std::string &name) {
   MG_ASSERT(replication_role_.load() == ReplicationRole::MAIN, "Only main instance can unregister a replica!");
+  if (ShouldStoreAndRestoreReplicas()) {
+    if (!kvstorage_->Delete(name)) {
+      spdlog::error("Error when removing replica {} from settings.", name);
```

Comment:
This is also interesting, however I am not sure what are the implications if we attach another replica that the user didn't want to.

Reply:
I think returning false is better than not doing anything. Otherwise user won't be aware of it (except if he pays attention to the error).
Not returning would mean Memgraph would pick up again the replica when starting again.